File: src/main/java/by/intexsoft/forum/entity/Role.java
package by.intexsoft.forum.entity;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Table;

/**
 * Role class describes the 'roles' table in DB.
 */
@Entity
@Table(name = "roles")
public class Role extends AbstractSystemEntity {
    private static final long serialVersionUID = -8986163028800488368L;

    @Column(unique = true, length = 10, nullable = false)
    public String title;
}

File: client/src/app/shared/entity/role.ts
import { AbstractSystemEntity } from './abstract/abstract-system-entity';

export class Role extends AbstractSystemEntity {
  title: string;
}

const user = 'USER';
const admin = 'ADMIN';
const manager = 'MANAGER';
const system = 'SYSTEM';

export class UserRole {
  static get USER(): string {
    return user;
  }

  static get ADMIN(): string {
    return admin;
  }

  static get MANAGER(): string {
    return manager;
  }

  static get SYSTEM(): string {
    return system;
  }
}
